File: api-java-sql/src/main/java/com/nebulahotel/repository/HospedesRepository.java
package com.nebulahotel.repository;

import com.nebulahotel.model.Hospedes;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface HospedesRepository extends JpaRepository<Hospedes, Integer> {

	Optional<Hospedes> findByEmail(String email);

	boolean existsByEmail(String email);

	boolean existsByCpf(String cpf);
}

File: api-kotlin-nosql/src/main/kotlin/com/nebulahotel/repository/HospedesRepository.kt
package com.nebulahotel.repository

import com.nebulahotel.model.Hospedes
import org.springframework.data.mongodb.repository.MongoRepository

interface HospedesRepository : MongoRepository<Hospedes, Int> {
    fun findByEmail(email: String): Hospedes?
    fun existsByEmail(email: String): Boolean
    fun existsByCpf(cpf: String): Boolean
}
